test/built-ins/Function/prototype/toString/built-in-function-object.js

```diff
@@ -6,43 +6,114 @@ description: >
   toString of built-in Function object
 info: |
   ...
-  If func is a Bound Function exotic object or a built-in Function object, then return an implementation-dependent String source code representation of func.
+  If func is a built-in Function object, then return an implementation-dependent String source code representation of func.
   The representation must have the syntax of a NativeFunction.
   ...
 
   NativeFunction:
-    function IdentifierName_opt ( FormalParameters ) { [ native code ] }
+    function NativeFunctionAccessor_opt IdentifierName_opt ( FormalParameters ) { [ native code ] }
+  NativeFunctionAccessor :
+    get
+    set
 
-includes: [nativeFunctionMatcher.js, wellKnownIntrinsicObjects.js]
-features: [arrow-function]
+includes: [nativeFunctionMatcher.js, fnGlobalObject.js]
+features: [arrow-function, Reflect]
 ---*/
 
-var visited = [];
-var verified = [];
-
-function visit(object) {
-  if (visited.includes(object)) {
+const visited = [];
+function visit(ns, path) {
+  if (visited.includes(ns)) {
     return;
   }
+  visited.push(ns);
 
-  visited.push(object);
-
-  if (typeof object === "function") {
-    assertNativeFunction(object);
-    verified.push(object.name);
+  if (typeof ns === 'function') {
+    assertNativeFunction(ns, path);
+  }
+  if (typeof ns !== 'function' && (typeof ns !== 'object' || ns === null)) {
+    return;
   }
 
-  for (var property of Object.getOwnPropertyNames(object)) {
-    if (typeof object[property] === "function" ||
-        typeof object[property] === "object") {
-      try {
-        visit(object[property], assertNativeFunction);
-      } catch(error) {
-        /* we don't actually want to do anything about failures here */
+  const descriptors = Object.getOwnPropertyDescriptors(ns);
+  Reflect.ownKeys(descriptors)
+    .forEach((name) => {
+      const desc = descriptors[name];
+      const p = typeof name === 'symbol'
+        ? `${path}[Symbol(${name.description})]`
+        : `${path}.${name}`;
+      if ('value' in desc) {
+        visit(desc.value, p);
+      } else {
+        visit(desc.get, p);
+        visit(desc.set, p);
       }
-    }
-  }
+    });
 }
 
-visit(WellKnownIntrinsicObjects.map(wkio => wkio.reference));
-assert.notSameValue(verified.length, 0);
+[
+  // Function Properties of the Global Object
```

Comment:
You might want to extract this list to constants in the helpers, kind of like what we had for the well-known intrinsics list before.